lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/Versions.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base;
+
+import io.trino.spi.connector.ConnectorContext;
+import io.trino.spi.connector.ConnectorFactory;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+
+public final class Versions
+{
+    private Versions() {}
+
+    public static void checkSpiVersion(ConnectorContext context, ConnectorFactory connectorFactory)
```

Comment:
How about adding a comment like:
```
Check if the SPI version of the Trino server matches exactly the SPI version the connector plugin was built for.
Using plugins built for a different version of Trino may fail if there are incompatible SPI changes,
but with an error message that does not directly indicate this.
Calling this in a plugin's connector factory will require building the plugin for every Trino version it will be used with. 
```

Reply:
thanks, added (slightly reworded)